src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -1235,9 +1236,20 @@ class ArnoldAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 				velocityGrids->hash( h );
 			}
 
-			h.append( velocityScale );
-			h.append( velocityFPS );
-			h.append( velocityOutlierThreshold );
+			if( velocityScale )
```

Comment:
There's a subtle bug here. I think we're unlikely to hit the conditions necessary to trigger it, but it's worth fixing, and worth knowing about in general given how much we rely on hashes everywhere. In the situations where only one of these three attributes is set, we will only append one value to the hash. In this case nothing in the hash distinguishes _which_ attribute was set, so we get hash collisions for the `velocityScale == X` and the `velocityFPS == X` cases. Does that make sense? The easiest solution is to always hash something for each attribute, maybe with an unconditional `h.append( velocityScale.get_value_or( default ) )`...

Reply:
Makes perfect sense, of course. I'll put up a fix later today! :)

Reply:
Thanks for catching that one, John. I've addressed it in f2ed774. If this is alright, I'll do another round of rebasing.

Reply:
Great - looks good. Thanks Matti! Once you've rebased could you just go ahead and merge?
Cheers...
John